tests/src/tools/ReadyToRun.SuperIlc/ApplicationSet.cs

```diff
@@ -0,0 +1,362 @@
+
+        private long _executionMilliseconds;
+
+        private long _buildMilliseconds;
+
+        public ApplicationSet(
+            IEnumerable<Application> applications,
+            IEnumerable<CompilerRunner> compilerRunners,
+            string coreRunPath,
+            string logPath)
+        {
+            _applications = applications;
+            _compilerRunners = compilerRunners;
+            _coreRunPath = coreRunPath;
+            _logPath = logPath;
+
+            _logWriter = new StreamWriter(_logPath);
+        }
+
+        public void Dispose()
+        {
+            _logWriter?.Dispose();
+        }
+
+        private void WriteJittedMethodSummary()
+        {
+            Dictionary<string, HashSet<string>>[] allMethodsPerModulePerCompiler = new Dictionary<string, HashSet<string>>[(int)CompilerIndex.Count];
+
+            foreach (CompilerRunner runner in _compilerRunners)
+            {
+                allMethodsPerModulePerCompiler[(int)runner.Index] = new Dictionary<string, HashSet<string>>();
+            }
+
+            foreach (Application app in _applications)
+            {
+                if (app.MainExecutable == null || app.Execution == null)
+                {
+                    continue;
+                }
+
+                Dictionary<string, HashSet<string>>[] appMethodsPerModulePerCompiler = new Dictionary<string, HashSet<string>>[(int)CompilerIndex.Count];
+
+                foreach (CompilerRunner runner in _compilerRunners)
+                {
+                    appMethodsPerModulePerCompiler[(int)runner.Index] = new Dictionary<string, HashSet<string>>();
+                    app.AddModuleToJittedMethodsMapping(allMethodsPerModulePerCompiler[(int)runner.Index], runner.Index);
+                    app.AddModuleToJittedMethodsMapping(appMethodsPerModulePerCompiler[(int)runner.Index], runner.Index);
+                }
+                app.WriteJitStatistics(appMethodsPerModulePerCompiler, _compilerRunners);
+            }
+
+            Application.WriteJitStatistics(_logWriter, allMethodsPerModulePerCompiler, _compilerRunners);
+        }
+
+        public bool Compile()
+        {
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            List<ProcessInfo> compilationsToRun = new List<ProcessInfo>();
+
+            foreach (Application application in _applications)
+            {
+                foreach (ProcessInfo[] compilation in application.Compilations)
+                {
+                    foreach (CompilerRunner runner in _compilerRunners)
+                    {
+                        ProcessInfo compilationProcess = compilation[(int)runner.Index];
+                        if (compilationProcess != null)
+                        {
+                            compilationsToRun.Add(compilationProcess);
+                        }
+                    }
+                }
+            }
+
+            Console.WriteLine();
```

Comment:
It would be nice to have consistency over our use of `Console.WriteLine` and `_logWriter.WriteLine`. Can everything just go through the LogWriter?

Reply:
I think this is good point and the current console output is far too noisy so I guess I'll just leave the bare minimum console output to enable watching progress for larger test suites and switch everything else to the file logs. I've also been thinking that I'll create another global log with concatenation of ILC and execution log for every test / runner combo. Today we're storing these per-test logs in individual files in the test output folders but you're right we'll want to start cleaning these up; and after all, it's sometimes easier to search a long log file than 10K tiny logs files scattered around the disk.

Reply:
Hopefully fixed in 3rd commit.